lib/_http_server.js

```diff
@@ -1038,9 +1038,17 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
     } else if (req.headers.expect !== undefined) {
       handled = true;
 
+      // From RFC 7230 5.4 https://datatracker.ietf.org/doc/html/rfc7230#section-5.4
+      // A server MUST respond with a 400 (Bad Request) status code to any
+      // HTTP/1.1 request message that lacks a Host header field
+      if (req.headers.host === undefined) {
+        res.writeHead(400);
+        res.end();
```

Comment:
It's maybe wise to reply with `Connection: close` in case the request is the result of some kind of desync attack.